api_tests/scripts/generic_request.py

```diff
@@ -125,12 +126,14 @@ def check_endpoint(self, verb: str, endpoint: str, expected_status_code: int,
         if type(expected_response) is list:
             assert self.verify_response_keys(response, expected_status_code, expected_keys=expected_response), \
                 f"UNEXPECTED RESPONSE {response.status_code}: {response.text}"
-            return True
+            return True if not return_response else response
```

Comment:
This smells like it should be two functions instead of one with a flag to change its return type